plugins/arm/semantics/aarch64-atomic.lisp

```diff
@@ -0,0 +1,41 @@
+(declare (context (target arm armv8-a+le)))
+
+(in-package aarch64)
+
+;;; ATOMIC OPERATIONS
+
+(defmacro CASordX (rs rt rn acquire-ordering release-ordering)
+  "(CASord*r set load store rs rt rn acquire-ordering release-ordering)
+   implements a generic compare-and-swap instruction on a X register.
+   acquire-ordering and release-ordering are booleans indicating whether
+   load-acquire and store-release ordering is to be enforced."
+   (let ((data (load-word rn)))
+    (when (acquire-ordering) (special :load-acquire))
+    (when (= data rs)
+      (when (release-odering) (special :store-release))
+      (store-word rn rt))
+    (set$ rs data)))
+
+(defun CASX   (rs rt rn) (CASordXr rs rt rn false false))
```

Comment:
maybe it should be `CASordX` without the trailing `r`?

Reply:
yep 👍